src/rust/rust-proto-new/Cargo.toml

```diff
@@ -28,3 +28,16 @@ quickcheck = "1.0"
 quickcheck_macros = "1.0"
 test-context = "0.1"
 tracing-subscriber = { version = "0.3", features = ["env-filter"] }
+
+[features]
+default = ["uid-allocator"]
+uid-allocator = []
+uid-allocator-messages = ["uid-allocator"]
+uid-allocator-server = ["uid-allocator-messages", "uid-allocator"]
+uid-allocator-client = ["uid-allocator-messages", "uid-allocator"]
+uid-allocator-full = [
+  "uid-allocator-messages",
+  "uid-allocator-client",
+  "uid-allocator-server",
+  "uid-allocator"
+]
```

Comment:
So, I kind of get what you're going for here, but wouldn't it make more sense to have the features be
uid-allocator
messages
server
client

?

Reply:
If we did that it would collide with other clients. ex: I add "graph-mutation-client" as a feature so that I can say "I just want that client" as opposed to all clients.

Reply:
I believe this could be done with like
```
#[cfg(all(feature = "uid-allocator", feature = "client"))]
```

but dont quote me

Reply:
just feels like a weird place/situation to reinvent the wheel